src/helpers/cauldron/lists/getMarketList.ts

```diff
@@ -0,0 +1,140 @@
+import { Contract, providers, utils } from "ethers";
+import { MulticallWrapper } from "ethers-multicall-provider";
+
+import cauldronsConfig from "@/utils/cauldronsConfig";
+import bentoBoxAbi from "@/utils/abi/bentoBox";
+
+import { getMaxToBorrow } from "@/helpers/cauldron/getMaxToBorrow";
+import { getTotalBorrowed } from "@/helpers/cauldron/getTotalBorrowed";
+
+const lensAddress = "0x73f52bd9e59edbdf5cf0dd59126cef00ecc31528";
+import lensAbi from "@/utils/abi/marketLens.js";
+
+type CauldronListItem = {
+  config: object;
+  interest: number;
+  tvl: string;
+  totalBorrowed: string;
+  MIMsLeftToBorrow: string;
+};
+
+export const getMarketList = async (
+  chainId: number,
+  provider: providers.BaseProvider
+): Promise<CauldronListItem[]> => {
+  const multicalProvider = MulticallWrapper.wrap(provider);
+
+  const configs: any[] = cauldronsConfig.filter(
+    (config) => config.chainId === chainId
+  );
+
+  const lensContract = new Contract(lensAddress, lensAbi, multicalProvider);
+
+  const cauldronContracts = configs.map((config: any) => {
+    return new Contract(
+      config.contract.address,
+      config.contract.abi,
+      multicalProvider
+    );
+  });
+
+  const boxAddresses = await Promise.all(
+    cauldronContracts.map((cauldron) => cauldron.bentoBox())
+  );
+
+  const boxContracts = boxAddresses.map((address: string) => {
+    return new Contract(address, bentoBoxAbi, multicalProvider);
+  });
+
+  const oracleRates = await Promise.all(
+    configs.map((config) =>
+      lensContract.getOracleExchangeRate(config.contract.address)
```

Comment:
Do you want to run a .catch here as well? Worried that one bad call will rekt the others...

`lensContract.getOracleExchangeRate(config.contract.address).catch(() => undefined)` or something?

Reply:
as far as i understand the only case where there could be an error here is a problem with rpc, but then all read methods will cause errors and this is quite difficult to trace and control

i also need to filter the cauldron and remove it from the list

i'm not sure that makes sense...

Reply:
You're probably right... We can test it and see.